src/pages/index.js

```diff
@@ -3,104 +3,248 @@ import FormValidator from "../components/FormValidator.js";
-    popupWithImage.open({ name: data.name, link: data.link });
+  console.log(data.name);
+  popupWithImage.open({ name: data.name, link: data.link });
+}
+
+function handleLikeButton(cardId, likeButton) {
+  if (!cardId) return;
+  const isLiked = likeButton.classList.contains("card__like-button_active");
+
+  api
+    .toggleLike(cardId, isLiked)
+    .then((updatedCard) => {
+      if (updatedCard && "isLiked" in updatedCard) {
+        likeButton.classList.toggle(
+          "card__like-button_active",
+          updatedCard.isLiked
+        );
+      }
+    })
+    .catch((error) => console.error("Error updating like:", error));
 }
 
 /*Event Handler*/
 
-function handleProfileEditSubmit(inputValue) {
-    userInfo.setUserInfo({
-        name: inputValue.title,
-        description: inputValue.description,
+function handleProfileEditSubmit(event) {
+  const inputValues = popupWithEditProfileForm.getInputValues(); // Get form values
+
+  if (!inputValues.name || !inputValues.about) {
+    return;
+  }
+
+  popupWithEditProfileForm.setLoadingText(true);
+
+  api
+    .updateUserInfo(inputValues.name, inputValues.about)
+    .then((updatedUser) => {
+      userInfo.setUserInfo(updatedUser);
+      popupWithEditProfileForm.close();
+    })
+    .catch((err) => console.error("Profile update failed:", err))
+    .finally(() => {
+      popupWithEditProfileForm.setLoadingText(false);
+    });
+}
+
+function handleAvatarUpdateSubmit(event) {
+  const inputvalues = popupWithEditAvatarForm.getInputValues(); // Get form values
+  const avatarUrl = inputvalues.avatar;
+
+  if (!avatarUrl) {
+    return;
+  }
+
+  popupWithEditAvatarForm.setLoadingText(true);
+  api
+    .updateProfileAvatar(avatarUrl)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+      popupWithEditAvatarForm.close();
+    })
+    .catch((err) => console.error("Profile update failed:", err))
+    .finally(() => {
+      popupWithEditAvatarForm.setLoadingText(false);
     });
-    popupWithEditProfileForm.close();
-    editFormValidator.disableSubmitButton();
 }
 
 function handleAddCardFormSubmit(inputValue) {
-    const cardData = {
-        name: inputValue.title,
-        link: inputValue.url,
-    };
-    renderCard(cardData);
-    popupWithAddCardForm.close();
-    addCardForm.reset();
-    addFormValidator.disableSubmitButton();
+  const cardData = {
+    name: inputValue.title,
+    link: inputValue.url,
+  };
+
+  popupWithAddCardForm.setLoadingText(true);
+  api
+    .addNewCard(cardData)
+    .then((newCard) => {
+      renderCard(newCard);
+    })
+    .finally(() => {
+      popupWithAddCardForm.setLoadingText(false);
+    });
+  popupWithAddCardForm.close();
+  addCardForm.reset();
+  addFormValidator.disableSubmitButton();
+}
 
+function openConfirmPopup(card) {
+  popupWithDeleteButton.setSubmitFunction(() => {
+    // this should be basically the handleDeleteCardConfirm logic
+    api
+      .deleteCardData(card._cardId)
```

Comment:
`_cardId`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.

And change it in the `Card` class as well